plugins/wazo-yealink/v84/tests/test_common.py

```diff
@@ -0,0 +1,429 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2021-2022 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from textwrap import dedent
+
+import pytest
+import six
+import six.moves as sm
+
+from hamcrest import (
+    assert_that,
+    equal_to,
+    has_entries,
+    has_properties,
+)
+from mock import MagicMock, Mock, patch, sentinel, PropertyMock
+from provd.devices.config import RawConfigError
+from provd.devices.pgasso import (
+    COMPLETE_SUPPORT,
+    FULL_SUPPORT,
+    IMPROBABLE_SUPPORT,
+    PROBABLE_SUPPORT,
+)
+from provd.tzinform import TimezoneNotFoundError, get_timezone_info
+
+from ..common import (
+    BaseYealinkHTTPDeviceInfoExtractor,
+    BaseYealinkPgAssociator,
+)
+
+if six.PY2:
+    FileNotFoundError = OSError
+
+BASE_TEST_LINES = """\
+linekey.1.type = 13
+linekey.1.line = 1
+linekey.1.value = test_speed_dial
+linekey.1.label = Test Speed dial
+
+linekey.2.type = 16
+linekey.2.line = 1
+linekey.2.value = test_blf
+linekey.2.label = Test blf
+linekey.2.extension = 1234
+"""
+
+TEST_LINES = BASE_TEST_LINES + """
+linekey.3.type = 10
+linekey.3.line = 1
+linekey.3.value = test_park
+linekey.3.label = Test Park
+"""
+
+
+class TestInfoExtraction(object):
+    @classmethod
+    def setup_class(cls):
+        cls.http_info_extractor = BaseYealinkHTTPDeviceInfoExtractor()
+
+    def _mock_request(self, ua=None, path=None):
+        request = MagicMock()
+        request.getHeader = MagicMock(return_value=ua)
+        request.path = path
+        return request
+
+    def test_http_ua_extractor_when_all_info(self):
+        ua_infos = {
+            'Yealink SIP-CP920 78.85.0.5 00:15:65:5e:16:7c': {
+                'vendor': 'Yealink',
+                'model': 'CP920',
+                'version': '78.85.0.5',
+                'mac': '00:15:65:5e:16:7c',
+            },
+            'Yealink SIP-T31G 124.85.257.55 80:5e:c0:d5:7d:72': {
+                'vendor': 'Yealink',
+                'model': 'T31G',
+                'version': '124.85.257.55',
+                'mac': '80:5e:c0:d5:7d:72',
+            },
+        }
+
+        for ua, info in ua_infos.items():
+            assert_that(
+                self.http_info_extractor._do_extract(self._mock_request(ua=ua)),
+                has_entries(info),
+            )
+
+    def test_http_ua_extractor_when_no_info(self):
+        assert self.http_info_extractor._extract_from_ua('') is None
```

Comment:
I know that `assert` is better when using pytest. However, we usually use `hamcrest` and its matchers for clarity purposes. The tests I wrote previously used hamcrest. It is possible to replace all hamcrest usage with assert if you want, or we can use hamcrest only, but not both.

Reply:
Oh ok. I thought hamcrest was a relic from the pre-pytest days. I was going to ask you. When I converted from test Unit test case I wasn't sure I needed to keep it.
If everyone wants to keep hamcrest in future I can try and convert them, but in my personal opinion I don't find it adds any clarity to tests and prefer just assert and the helpers directly provided by pytest. I find that `assert` is simple and clear and doesn't requires knowledge of the library to know what the functions do.  However, maybe there is an advantage I'm unfamiliar with.